scripts/metrics/common/config.py

```diff
@@ -0,0 +1,79 @@
+"""
+Copyright (c) Contributors to the Open 3D Engine Project.
+For complete copyright and license terms please see the LICENSE at the root of this distribution.
+
+SPDX-License-Identifier: Apache-2.0 OR MIT
+
+Helper file for assisting in building workspaces and setting up LTT with the current Lumberyard environment.
+"""
+
+import os
+import sys
+
+import yaml
+
+from common.exception import MetricsExn
+
+config = {}
+
+
+class ConfigExn(MetricsExn):
+    pass
+
+
+def load_config(local_path, override=None):
+    global config
+
+    if override is None:
+        override = {}
+
+    try:
+        with open(os.path.join(sys.path[0], local_path), "r") as file:
+            config = yaml.safe_load(file)
+    except FileNotFoundError:
+        raise ConfigExn(f"Failed to load configuration from local path '{local_path}'") from None
+
+    for path, value in override:
+        path_elements = path.split(".")
+        target = config
+
+        while len(path_elements) > 0:
+            key = path_elements.pop(0)
+
+            if isinstance(target, dict) and key in target:
+                if len(path_elements) == 0:
+                    if isinstance(target[key], (int, float, str, bool)):
+                        try:
+                            target[key] = type(target[key])(value)
```

Comment:
Why is it necessary to coerce this value to be of the same type as the previous value?

Reply:
This should prevent any string/int/float conversions from the config override